chain/tests/test_pibd_copy.rs

```diff
@@ -0,0 +1,233 @@
+			}
+		}
+
+		// Finalize segmenter bitmap, which means we've recieved all bitmap MMR chunks and
+		// Are ready to use it to validate outputs
+		desegmenter.finalize_bitmap().unwrap();
+
+		// OUTPUTS  - Read + Validate
+		let identifier_iter = SegmentIdentifier::traversal_iter(
+			horizon_header.output_mmr_size,
+			target_segment_height,
+		);
+
+		for sid in identifier_iter {
+			debug!("Getting output segment with Segment Identifier {:?}", sid);
+			let (output_segment, bitmap_root_hash) = segmenter.output_segment(sid).unwrap();
+			debug!(
+				"Output segmenter reports bitmap hash is {:?}",
+				bitmap_root_hash
+			);
+			// Add segment to desegmenter / validate
+			if let Err(e) = desegmenter.add_output_segment(output_segment) {
+				panic!("Unable to add output segment: {}", e);
+			}
+		}
+
+		// PROOFS  - Read + Validate
+		let identifier_iter = SegmentIdentifier::traversal_iter(
+			horizon_header.output_mmr_size,
+			target_segment_height,
+		);
+
+		for sid in identifier_iter {
+			debug!(
+				"Getting rangeproof segment with Segment Identifier {:?}",
+				sid
+			);
+			let rangeproof_segment = segmenter.rangeproof_segment(sid).unwrap();
+			// Add segment to desegmenter / validate
+			if let Err(e) = desegmenter.add_rangeproof_segment(rangeproof_segment) {
+				panic!("Unable to add rangeproof segment: {}", e);
+			}
+		}
+
+		// KERNELS - Read + Validate
+		let identifier_iter = SegmentIdentifier::traversal_iter(
+			horizon_header.kernel_mmr_size,
+			target_segment_height,
+		);
+
+		for sid in identifier_iter {
+			debug!("Getting kernel segment with Segment Identifier {:?}", sid);
+			let kernel_segment = segmenter.kernel_segment(sid).unwrap();
+			if let Err(e) = desegmenter.add_kernel_segment(kernel_segment) {
+				panic!("Unable to add kernel segment: {}", e);
+			}
+		}
+
+		let dest_txhashset = dest_chain.txhashset();
+		debug!("Dest TxHashset Roots: {:?}", dest_txhashset.read().roots());
+	}
+}
+
+#[test]
+fn test_pibd_copy_sample() {
+	util::init_test_logger();
+	// Note there is now a 'test' in grin_wallet_controller/build_chain
+	// that can be manually tweaked to create a
+	// small test chain with actual transaction data
+
+	// Test on uncompacted and non-compacted chains
+	let src_root_dir = format!("./tests/test_data/chain_raw");
+	let dest_root_dir = format!("./tests/test_output/.segment_copy");
+	clean_output_dir(&dest_root_dir);
+	test_pibd_copy_impl(true, &src_root_dir, &dest_root_dir);
+	let src_root_dir = format!("./tests/test_data/chain_compacted");
+	clean_output_dir(&dest_root_dir);
+	test_pibd_copy_impl(true, &src_root_dir, &dest_root_dir);
+	clean_output_dir(&dest_root_dir);
+}
+
+#[test]
+#[ignore]
+// As above, but run on a real instance of a chain pointed where you like
+fn test_pibd_copy_real() {
+	util::init_test_logger();
+	// if testing against a real chain, insert location here
+	let src_root_dir = format!("/Users/yeastplume/Projects/grin_project/server/chain_data");
```

Comment:
This is not gonna work on my machine:-(

Reply:
it's not meant to, it's meant for running a manual test against a live chain instance (so while testing you modify the path).. I'll make this more clear in comments